crates/forge/src/runner.rs

```diff
@@ -472,9 +472,20 @@ impl<'a> FunctionRunner<'a> {
             return self.result;
         }
 
+        let test_fail_deprecation = |should_fail: bool| {
```

Comment:
i think we want to deprecate with a warning first for 0.3?

Reply:
Makes sense! https://github.com/foundry-rs/foundry/pull/9581